File: Leetcode/C++/28. Find the Index of the First Occurrence in a String/solution.cpp
// https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/description/

#include <string>

using namespace std;

class Solution
{
public:
    int strStr(string haystack, string needle)
    {
        for (int left = 0; left < haystack.size(); left++)
        {
            int right;
            for (right = 0; right < needle.size(); right++)
            {
                if (haystack[left + right] != needle[right])
                {
                    break;
                }
            }
            if (right == needle.size())
            {
                return left;
            }
        }
        return -1;
    }
};

File: Leetcode/Python/28. Find the Index of the First Occurrence in a String/solution.py
# https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/description/


class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        for left in range(len(haystack) - len(needle) + 1):
            for idx in range(len(needle)):
                if haystack[left + idx] != needle[idx]:
                    break
            else:
                return left
        return -1


class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        return haystack.find(needle)
